ark/mibi/tiling_utils_test.py

```diff
@@ -0,0 +1,550 @@
+
+        # write moly point
+        sample_moly_path = os.path.join(temp_dir, 'moly_point.json')
+        with open(sample_moly_path, 'w') as moly:
+            json.dump(sample_moly_point, moly)
+
+        # run tiling parameter setting process with predefined user inputs
+        sample_tiling_params, moly_point = tiling_utils.set_tiling_params(
+            sample_fov_list_path, sample_moly_path
+        )
+
+        # assert the fovs in the tiling params are the same as in the original fovs list
+        assert sample_tiling_params['fovs'] == sample_fovs_list['fovs']
+
+        # assert region start x and region start y values are correct
+        region_start_x = sample_tiling_params['region_start_x']
+        region_start_y = sample_tiling_params['region_start_y']
+        fov_0 = sample_fovs_list['fovs'][0]
+        fov_1 = sample_fovs_list['fovs'][1]
+
+        assert region_start_x[0] == fov_0['centerPointMicrons']['x']
+        assert region_start_x[1] == fov_1['centerPointMicrons']['x']
+        assert region_start_y[0] == fov_0['centerPointMicrons']['y']
+        assert region_start_y[1] == fov_1['centerPointMicrons']['y']
+
+        # assert fov_num_x and fov_num_y are all set to 1
+        assert all(nx == 1 for nx in sample_tiling_params['fov_num_x'])
+        assert all(ny == 1 for ny in sample_tiling_params['fov_num_y'])
+
+        # assert x_fov_size and y_fov_size are all set to 1
+        assert all(sx == 1 for sx in sample_tiling_params['x_fov_size'])
+        assert all(sy == 1 for sy in sample_tiling_params['y_fov_size'])
+
+        # assert randomize is set to 1 for both fovs
+        assert all(r == 1 for r in sample_tiling_params['randomize'])
+
+        # assert moly run is set to 1
+        assert sample_tiling_params['moly_run'] == 1
+
+        # assert moly interval is set to 1
+        assert sample_tiling_params['moly_interval'] == 1
+
+
+def test_create_tiled_regions():
+    # set tiling_params dict
+    sample_tiling_params = {
+        'fovFormatVersion': '1.5',
+        'fovs': [
+            {
+                'scanCount': 1,
+                'centerPointMicrons': {
+                    'x': 0, 'y': 0
+                },
+                'timingChoice': 7,
+                'frameSizePixels': {
+                    'width': 2048, 'height': 2048
+                },
+                'imagingPreset': {
+                    'preset': 'Normal',
+                    'aperture': '2',
+                    'displayName': 'Fine',
+                    'defaults': {
+                        'timingChoice': 7
+                    }
+                },
+                'sectionId': 8201,
+                'slideId': 5931,
+                'name': 'TheFirstFOV',
+                'timingDescription': '1 ms'
+            },
+            {
+                'scanCount': 1,
+                'centerPointMicrons': {
+                    'x': 100, 'y': 100
+                },
+                'timingChoice': 7,
```

Comment:
I think we can move this stuff to a helper function that creates a JSON file. Can probably combine this with the JSON file in previous function